### DISCUSSION TITLE
maxFeePerGas requirement in 4.x

### DISCUSSION THREAD

**Discussion by t-vila** [ethereum/web3.js#6188] [General]
Hey all,

As far as I remember `maxFeePerGas` was optional in 1.x and calculated from the `baseFee` - in 4.x I get `MissingGasError` if I don't specify it. It looks like it's expected https://github.com/web3/web3.js/blob/f8a2533c2b09ce0a62f8414f2f6eed83ab78ca1f/packages/web3-errors/src/errors/transaction_errors.ts#L315

For instance this:

```
const tx = {
    from: signer.address,
    to: "...",
    value: web3.utils.toWei('0.0001', 'ether'),
    gas: limit,
    nonce: await web3.eth.getTransactionCount(signer.address),
    maxPriorityFeePerGas: web3.utils.toWei('3', 'gwei'),
    //maxFeePerGas: web3.utils.toWei('3', 'gwei'),
    chainId: 11155111,
    type: 0x2
  };
```
Will throw:

`MissingGasError: Invalid value given "gas: 0x5208, gasPrice: undefined, maxPriorityFeePerGas: 0xb2d05e00, maxFeePerGas: undefined". Error: "gas" is missing.`

**Comment by luu-alex** [comment]
Hi there, thank you for reporting this. can you make this an issue

**Reply by t-vila** [reply]
hey, yes ofc, will do - thanks !
